### PR TITLE
Backport of master changes to eventloop-2.0

### PR DESCRIPTION
**Pull request by vberger**
Like the previous one (#833) I cherry-picked the commits.

### CODE REVIEW

**Comment by ZeGentzy** [comment]
https://github.com/rust-windowing/winit/pull/831 ?

**Comment by vberger** [comment]
It is already on the the `eventloop-2.0` branch : https://github.com/rust-windowing/winit/commit/47194b5f3cb74c41d435d48312172e6bc0839da2

File: src/platform_impl/windows/window.rs [hunk 1]
```diff
@@ -399,6 +399,12 @@ impl Window {
         });
     }
 
+    #[inline]
+    pub fn get_fullscreen(&self) -> Option<RootMonitorHandle> {
+        let window_state = self.window_state.lock().unwrap();
```
Comment (aloucks) [inline src/platform_impl/windows/window.rs]: Looks like windows was added to the list of targets that are using `parking_lot`, which doesn't have `unwrap`.

Comment (vberger) [reply src/platform_impl/windows/window.rs]: Oh, right. Will correct that, thanks.

**Comment by Osspial** [comment]
Whoops, it looks like when I merged this, I unintentionally messed up the commit history! I'll get on fixing that.

**Pull request merged**